packages/features/ee/teams/components/MemberInvitationModal.tsx

```diff
@@ -389,8 +379,36 @@ export default function MemberInvitationModal(props: MemberInvitationModalProps)
                   type="button"
                   color="minimal"
                   variant="icon"
-                  onClick={() => {
-                    createInviteMutation.mutate({ teamId: props.teamId, token: props.token });
+                  onClick={async function () {
+                    try {
+                      // Required for Safari but also works on Chrome
+                      // Credits to https://wolfgangrittner.dev/how-to-use-clipboard-api-in-firefox/
+                      if (typeof ClipboardItem) {
+                        const inviteLinkClipbardItem = new ClipboardItem({
+                          "text/plain": new Promise(async (resolve) => {
+                            // Instead of doing async work and then writing to clipboard, do async work in clipboard API itself
+                            const { inviteLink } = await createInviteMutation.mutateAsync({
```

Comment:
Spent quite some time figuring out two promises to work together with closures going on. In the end discovered that there is mutateAsync to make it simpler 😅